mining/src/manager.rs

```diff
@@ -210,24 +210,66 @@ impl MiningManager {
     }
 
     /// Returns realtime feerate estimations based on internal mempool state with additional verbose data
-    pub(crate) fn get_realtime_feerate_estimations_verbose(&self) -> FeeEstimateVerbose {
+    pub(crate) fn get_realtime_feerate_estimations_verbose(
+        &self,
+        consensus: &dyn ConsensusApi,
+        prefix: kaspa_addresses::Prefix,
+    ) -> MiningManagerResult<FeeEstimateVerbose> {
         let args = FeerateEstimatorArgs::new(self.config.network_blocks_per_second, self.config.maximum_mass_per_block);
         let network_mass_per_second = args.network_mass_per_second();
         let mempool_read = self.mempool.read();
         let estimator = mempool_read.build_feerate_estimator(args);
         let ready_transactions_count = mempool_read.ready_transaction_count();
         let ready_transaction_total_mass = mempool_read.ready_transaction_total_mass();
         drop(mempool_read);
-        FeeEstimateVerbose {
+        let mut resp = FeeEstimateVerbose {
             estimations: estimator.calc_estimations(self.config.minimum_feerate()),
             network_mass_per_second,
             mempool_ready_transactions_count: ready_transactions_count as u64,
             mempool_ready_transactions_total_mass: ready_transaction_total_mass,
-            // TODO: Next PR
+
             next_block_template_feerate_min: -1.0,
             next_block_template_feerate_median: -1.0,
             next_block_template_feerate_max: -1.0,
+        };
+        // calculate next_block_template_feerate_xxx
+        {
+            let script_public_key = kaspa_txscript::pay_to_address_script(&kaspa_addresses::Address::new(
+                prefix,
+                kaspa_addresses::Version::PubKey,
+                &[0u8; 32],
+            ));
+            let miner_data: MinerData = MinerData::new(script_public_key, vec![]);
+
+            let BlockTemplate { block: kaspa_consensus_core::block::MutableBlock { transactions, .. }, calculated_fees, .. } =
+                self.get_block_template(consensus, &miner_data)?;
+            // don't consider coinbase tx
+            if transactions.len() < 2 {
+                return Ok(resp);
+            }
+            let mut fees_and_masses = transactions
+                .iter()
+                // skip coinbase tx
+                .skip(1)
+                .map(Transaction::mass)
+                .zip(calculated_fees)
```

Comment:
you zip `fee` to `mass` (`mass` first), but then use `(fee, mass)` tuples..

Reply:
maybe move this piece of logic (i.e., the logic of getting min/max/median from a vec of txs and a vec of fees) to a separate method and add a short test verifying it

Reply:
[8f5f802](https://github.com/kaspanet/rusty-kaspa/pull/523/commits/8f5f802fdaee63737de3435be5dc81c8c66ce84a)